zimbra/src/main/java/fr/openent/zimbra/controllers/ZimbraController.java

```diff
@@ -1268,17 +1265,27 @@ public void zimbraEventBusHandler(Message<JsonObject> message) {
                     UserUtils.getUserInfos(eb, userId, user -> {
                         Boolean hasExpertRight = WorkflowActionUtils.hasRight(user, WorkflowActions.EXPERT_ACCESS_RIGHT.toString());
                         if (Boolean.TRUE.equals(hasExpertRight)) {
-                            calendarServiceImpl.getICal(user)
-                                    .onSuccess(ical -> {
-                                        ical = ical.replaceAll("\\\r\\\n", "\n");
-                                        ical = ical.replaceAll(";TZID\\=.*\\/.*\\\"", "");
-                                        message.reply(new JsonObject().put(Field.STATUS, Field.OK).put(Field.RESULT, new JsonObject().put(Field.ICS, ical)));
-                                    })
+                            JsonObject icalRequest = requestHelper.createICalRequest();
+                            queueServiceImpl.putRequestInQueue(user, icalRequest)
+                                    .onSuccess(result -> message.reply(new JsonObject().put(Field.STATUS, Field.OK)))
                                     .onFailure(error -> {
-                                        String errMessage = String.format("[Zimbra@%s::zimbraEventBusHandler]: get-platform-ics : error during ical retrieval: %s",
-                                                this.getClass().getSimpleName(), error.getMessage());
-                                        EventBusHelper.eventBusError(errMessage, "zimbra.ics.retrieval.error", message);
+                                        String errMessage = String.format("[Zimbra@%s::zimbraEventBusHandler]: get-platform-ics : error when putting " +
+                                                        "ical request in queue: %s", this.getClass().getSimpleName(), error.getMessage());
+                                        EventBusHelper.eventBusError(errMessage, "zimbra.ical.request.queue.error", message);
                                     });
+                            //todo to be used in ical worker
+//                            message.reply(new JsonObject().put(Field.STATUS, Field.OK).put(Field.RESULT, new JsonObject().put(Field.STATUS, Field.PENDING)));
```

Comment:
used in [MZI-162](https://jira.support-ent.fr/browse/MZI-162)

Reply:
Evite de balancer du code commenter dans tes commits

Reply:
Ce code était déjà présent dans la [MZI-105](https://jira.support-ent.fr/browse/MZI-105) mais avec la refacto en cours je l'ai commenté pour pouvoir l'utiliser plus tard dans un worker qui n'est pas encore créé